### PR TITLE
Keep original stack trace

### LINKED ISSUE #177: Stacktraces don't show original caller
Since we hand over the request execution to a thread pool already, we loose the original call site if an exception occurs. Makes it a lot harder to debug an issue than it should be.

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #177

### CODE REVIEW

File: riptide-core/src/main/java/org/zalando/riptide/NoWildcardException.java [hunk 1]
```diff
@@ -2,4 +2,13 @@
 
 public final class NoWildcardException extends RuntimeException {
 
+    /**
+     * We don't care for the stack trace, and since it's expensive to calculate we just use the same instance
+     * all the time. It's exclusively used to jump around the stack.
```
Comment (lukasniemeier-zalando) [inline riptide-core/src/main/java/org/zalando/riptide/NoWildcardException.java]: `we just use the same instance all the time` is not true anymore as it seems

File: riptide-core/src/main/java/org/zalando/riptide/Requester.java [hunk 2]
```diff
@@ -83,36 +87,9 @@ public final Requester contentType(final MediaType contentType) {
...
-                        future.completeExceptionally(e);
-                    }
-                }, future::completeExceptionally);
-
-                return future;
-            }
+        final Exception exceptionWithOriginalStackTrace = new Exception();
 
-        };
+        return new ResponseDispatcher(worker, listenable, exceptionWithOriginalStackTrace);
```
Comment (lukasniemeier-zalando) [inline riptide-core/src/main/java/org/zalando/riptide/Requester.java]: Why not pass only `new Throwable().getStackTrace()` here?

File: riptide-core/src/main/java/org/zalando/riptide/NoWildcardException.java (hunk 1 above)
Comment (whiskeysierra) [inline riptide-core/src/main/java/org/zalando/riptide/NoWildcardException.java]: Correct, fixed.

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**